src/interactive-window/editor-integration/codeGenerator.ts

```diff
@@ -53,13 +54,17 @@ export class CodeGenerator implements IInteractiveWindowCodeGenerator {
         this.executionCount = 0;
     }
 
-    public generateCode(metadata: Pick<InteractiveCellMetadata, 'interactive' | 'id'>, debug: boolean) {
+    public generateCode(
+        metadata: Pick<InteractiveCellMetadata, 'interactive' | 'id'>,
+        debug: boolean,
+        usingJupyterDebugProtocol?: boolean
```

Comment:
Added argument to decide what kind of debugging code is to be added.
Didn't want to pass in IKernel or other stuff. 
the desire is that we'll switch over to the new Jupyter protocol debugger for everything and deprecate the old IW debugger code